aten/src/ATen/native/cuda/CUDALoops.cuh

```diff
@@ -266,7 +312,8 @@ void gpu_kernel_impl(TensorIterator& iter, const func_t& f) {
       strides[i] = inner_strides[i];
     }
 
-    if (needs_dynamic_casting<func_t>::check(iter)) {
+    // TODO: can non_c10_complex go through the other path?  Need to verify.
```

Comment:
There is a `ROCmLoops.cuh` which is a copy-paste of an old version of CUDALoops.cuh, due to compiler bugs in ROCm is blocking the vectorization of GPU loops. It also contains a similar logic. And if are trying to make `needs_dynamic_casting<func_t>::check(iter)` return false for `thrust<-->c10<-->std` casts, you might need to add `|| non_c10_complex` to `ROCmLoops.cuh` as well.

Reply:
oh I see, thanks for pointing this out.